### PR TITLE
Add draft of the Swift 5.7 release blog post.

### PR DESCRIPTION
**Pull request by hborla**
The content in this post has been gathered from various people who have contributed to Swift 5.7. All feedback is welcome!

### CODE REVIEW

File: _posts/2022-09-06-swift-5.7-released.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+* SE-0333: [Expand usability of withMemoryRebound](https://github.com/apple/swift-evolution/blob/main/proposals/0333-with-memory-rebound.md)
+* SE-0334: [Pointer API Usability Improvements](https://github.com/apple/swift-evolution/blob/main/proposals/0334-pointer-usability-improvements.md)
+* SE-0349: [Unaligned Loads and Stores from Raw Memory](https://github.com/apple/swift-evolution/blob/main/proposals/0349-unaligned-loads-and-stores.md)
+
+## Developer Experience
+
+### New Generics Implementation
+
+In addition to the above language improvements for working with generics, the type checker’s generics implementation has rewritten from scratch with improvements to correctness and performance.
```
Comment (cthielen) [inline _posts/2022-09-06-swift-5.7-released.md:63]: "has written" to "has been rewritten"?

File: _posts/2022-09-06-swift-5.7-released.md [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+The new generics implementation also improves performance. With certain configurations of protocols and associated types, type checking time would scale exponentially in Swift 5.6, but is now linear in Swift 5.7.
+
+### Automatic Reference Counting Improvements
+
+In Swift 5.7, ARC behavior is more predictable, user-friendly, and performant by specifying new rules for when optimization is allowed to shorten the lifetime of variables. To enforce the rules, the compiler adopted a new internal representation that tracks the lexical scope of each variable. This involved updating existing optimizations and implementing several new optimizations. Now, the most common programming patterns that depend on extended variable lifetimes are safe without requiring programmers to explicitly use `withExtendedLifetime()`. This protects you from difficult to diagnose lifetime bugs that only appear at runtime in optimized builds. It also allows the introduction of more powerful optimization without breaking existing source.
```
Comment (cthielen) [inline _posts/2022-09-06-swift-5.7-released.md:71]: "by specifying new rules to shorten the lifetime of variables when optimization is allowed"

Comment (cthielen) [inline _posts/2022-09-06-swift-5.7-released.md:71]: "difficult-to-diagnose"

File: _posts/2022-09-06-swift-5.7-released.md [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+
+### Code Completion
+
+Code completion of function call arguments, variables and global functions is now tightly integrated into Swift’s type checker. This allows code completion to provide more accurate results inside ambiguous code or code with errors.
```
Comment (cthielen) [inline _posts/2022-09-06-swift-5.7-released.md:75]: "function call arguments, variables, and global functions" (add in the last, serial comma for consistency)

File: _posts/2022-09-06-swift-5.7-released.md [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+
+If completing after the `+` the following example, code completion now reports `int` and `string` as matching the surrounding context, allowing editors to rank these results higher than `array`.
+
+```
```
Comment (cthielen) [inline _posts/2022-09-06-swift-5.7-released.md:79]: We need "```swift" on the code samples to get proper highlighting with Jekyll.

File: _posts/2022-09-06-swift-5.7-released.md [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+func add(_ firstInt: Int, secondInt: Int) {}
+func add(_ firstString: String, secondString: String) {}
+add(1, )
+```
+
+## Ecosystem
+
+### SourceKit-LSP
+
+SourceKit-LSP received several improvements to support the release of the [Swift for Visual Studio Code Plugin](https://forums.swift.org/t/introducing-swift-for-visual-studio-code/54246).
```
Comment (cthielen) [inline _posts/2022-09-06-swift-5.7-released.md:101]: Swap "plugin" for "extension" to match VS Code terminology.

File: _posts/2022-09-06-swift-5.7-released.md [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+
+If multiple SwiftPM projects are opened in the same Visual Studio Code workspace, SourceKit-LSP can manage them in a single server instance. The language server no longer needs to be restarted when switching between files from different SwiftPM projects, which reduces the delay until semantic functionality becomes available for a file.
+
+### Swift Package Manager
+
+The following [Swift Evolution](https://github.com/apple/swift-evolution) proposals for SwiftPM were accepted and [implemented in Swift 5.7](https://apple.github.io/swift-evolution/#?version=5.7):  
+
+* SE-0292: [Package Registry Service](https://github.com/apple/swift-evolution/blob/main/proposals/0292-package-registry-service.md)
+* SE-0303: [Build tool plugins](https://github.com/apple/swift-evolution/blob/main/proposals/0303-swiftpm-extensible-build-tools.md) and SE-0332: [Comannd Plugins](https://github.com/apple/swift-evolution/blob/main/proposals/0332-swiftpm-command-plugins.md)  which first introduced in Swift 5.6 have been further refined and made available via Xcode.
```
Comment (cthielen) [inline _posts/2022-09-06-swift-5.7-released.md:112]: "which were first "

File: _posts/2022-09-06-swift-5.7-released.md [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+* SE-0339: [Module Aliasing For Disambiguation](https://github.com/apple/swift-evolution/blob/main/proposals/0339-module-aliasing-for-disambiguation.md)
+
+### Distributed Actor Library
+
+Based on SE-0336: [Distributed Actor Isolation](https://github.com/apple/swift-evolution/blob/main/proposals/0336-distributed-actor-isolation.md), a [server library](https://github.com/apple/swift-distributed-actors) for building distributed systems in Swift has been made open source. The library provides peer-to-peer cluster actor system implementation, including cluster management, failure detection, and integration with service discovery.
```
Comment (cthielen) [inline _posts/2022-09-06-swift-5.7-released.md:117]: "The library provides a peer-to-peer" ?

File: _posts/2022-09-06-swift-5.7-released.md [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+
+### Windows Platform
+
+The Windows support continues to be refined with each release of Swift. The Windows toolchain has migrated and fully adopted swift-driver as the compiler driver and the C++ driver is not packaged or provided as a fallback. The installer received improvements for robustness of deploying the module-maps and packaging of the files which reduced the size of the installer download. The API digester and other tools have been added in the new smaller download. The SDK layout has been further streamlined by removing unnecessary path components. The developer libraries were restructured to allow multi-architectural installations as work continues to support ARM64 and X86 targets in addition to X64. Some initial support for static Swift libraries was implemented which covers libraries other than the Swift standard library, and can be used with build tools such as CMake and Bazel on Windows.
```
Comment (cthielen) [inline _posts/2022-09-06-swift-5.7-released.md:121]: More natural to say "Windows support" without the 'The' in the first sentence.

Comment (cthielen) [inline _posts/2022-09-06-swift-5.7-released.md:121]: "toolchain has migrated to fully adopt swift-driver" ?

Comment (cthielen) [inline _posts/2022-09-06-swift-5.7-released.md:121]: "the C++ driver is no longer packaged nor provided"

Comment (cthielen) [inline _posts/2022-09-06-swift-5.7-released.md:121]: Should this paragraph instead be a single introductory sentence followed by a list? As a paragraph, it reads a little bit like a series of unrelated changes.

File: _posts/2022-09-06-swift-5.7-released.md [hunk 9]
```diff
@@ -0,0 +1,143 @@
...
+
+**Pointer Usability**
+
+* SE-0333: [Expand usability of withMemoryRebound](https://github.com/apple/swift-evolution/blob/main/proposals/0333-with-memory-rebound.md)
+* SE-0334: [Pointer API Usability Improvements](https://github.com/apple/swift-evolution/blob/main/proposals/0334-pointer-usability-improvements.md)
+* SE-0349: [Unaligned Loads and Stores from Raw Memory](https://github.com/apple/swift-evolution/blob/main/proposals/0349-unaligned-loads-and-stores.md)
+
+## Developer Experience
+
+### New Generics Implementation
```
Comment (cthielen) [inline _posts/2022-09-06-swift-5.7-released.md:61]: Is there a SE or series of SEs we can refer to in each of the 'Developer Experience' sub-headers?

File: _posts/2022-09-06-swift-5.7-released.md [hunk 10]
```diff
@@ -0,0 +1,143 @@
...
+author: [hborla]
+---
+
+Swift 5.7 is now officially released! Swift 5.7 includes major enhancements to concurrency, string processing, the generics system, and other usability improvements in the Swift language and ecosystem.
+
+Swift 5.7 is the culmination of countless contributions from members across the Swift community. Thank you to everyone in the community for your Swift Forum discussions, bug reports, pull requests, educational content, and much more!
+
+If you’re new to Swift, [The Swift Programming Language](https://docs.swift.org/swift-book/) is the official Swift guide and has been updated for version 5.7. The Swift community also maintains a number of [translations](https://www.swift.org/documentation/#translations). It is also available for free on the [Apple Books store](https://itunes.apple.com/us/book/the-swift-programming-language/id881256329?mt=11).
+
+## Language and Standard Library
```
Comment (cthielen) [inline _posts/2022-09-06-swift-5.7-released.md:15]: What do folks think about putting the list of SEs closer to the bottom of the post? It feels a little bit like we're throwing an index at readers toward the top of the post. The post might be better as "Here's what we're announcing … here's a closer look at the more notable changes … ecosystem specific changes … how to download … complete index of SE changes".

Comment (alexandersandberg) [reply _posts/2022-09-06-swift-5.7-released.md:15]: Makes sense to me to put the complete list at the bottom. 👍

File: _posts/2022-09-06-swift-5.7-released.md [hunk 11]
```diff
@@ -0,0 +1,143 @@
...
+add(1, )
+```
+
+## Ecosystem
+
+### SourceKit-LSP
+
+SourceKit-LSP received several improvements to support the release of the [Swift for Visual Studio Code Plugin](https://forums.swift.org/t/introducing-swift-for-visual-studio-code/54246).
+
+Compiler arguments are now recomputed after changes to `Package.swift`, `compile_commands.json` or `compile_flags.txt`. This ensures that semantic functionality, such as code completion or live issues, continues to work correctly after these changes.
```
Comment (alexandersandberg) [inline _posts/2022-09-06-swift-5.7-released.md:103]:
```suggestion
Compiler arguments are now recomputed after changes to `Package.swift`, `compile_commands.json`, or `compile_flags.txt`. This ensures that semantic functionality, such as code completion or live issues, continues to work correctly after these changes.
```

(add serial comma for consistency)

File: _posts/2022-09-06-swift-5.7-released.md (hunk 8 above)
Comment (alexandersandberg) [inline _posts/2022-09-06-swift-5.7-released.md:121]: This sentence is a bit difficult to read.

_"The installer received improvements for robustness of deploying the module-maps and packaging of the files which reduced the size of the installer download."_

Would something like this be better?

_"The installer's robustness of module-maps deployment and file packaging was improved which reduced the download size."_

Comment (hborla) [reply _posts/2022-09-06-swift-5.7-released.md:121]: That's a good idea, I'll try that. @compnerd let me know if you have any ideas too, or if there are any particular changes that should be emphasized.

File: _posts/2022-09-06-swift-5.7-released.md (hunk 9 above)
Comment (hborla) [reply _posts/2022-09-06-swift-5.7-released.md:61]: No, compiler implementation changes that don't change language semantics don't go through the Swift evolution process. Same with SourceKit changes (for the Code Completion section). There is a forum post for the New Generics Implementation work, but it's more focused on the implementation approach rather than the impact on Swift programmers (https://forums.swift.org/t/the-requirement-machine-a-new-generics-implementation-based-on-term-rewriting/55601)

File: _posts/2022-09-06-swift-5.7-released.md (hunk 8 above)
Comment (hborla) [reply _posts/2022-09-06-swift-5.7-released.md]: How does it read now?

File: _posts/2022-09-06-swift-5.7-released.md (hunk 10 above)
Comment (hborla) [reply _posts/2022-09-06-swift-5.7-released.md:15]: 👍🏻 I'm working on this

Comment (hborla) [reply _posts/2022-09-06-swift-5.7-released.md:15]: How does it look now?

**Review by tomerd — APPROVED** [review]

**Comment by tomerd** [comment]
@swift-ci test

File: _posts/2022-09-06-swift-5.7-released.md [hunk 12]
```diff
@@ -0,0 +1,161 @@
...
+
+### Code Completion
+
+Code completion of function call arguments, variables, and global functions is now tightly integrated into Swift’s type checker. This allows code completion to provide more accurate results inside ambiguous code or code with errors.
+
+If completing after the `+` the following example, code completion now reports `int` and `string` as matching the surrounding context, allowing editors to rank these results higher than `array`.
```
Comment (alexandersandberg) [inline _posts/2022-09-06-swift-5.7-released.md:45]:
```suggestion
If completing after the `+` in the following example, code completion now reports `int` and `string` as matching the surrounding context, allowing editors to rank these results higher than `array`.
```

I feel like an "in" is missing here?

File: _posts/2022-09-06-swift-5.7-released.md (hunk 8 above)
Comment (alexandersandberg) [reply _posts/2022-09-06-swift-5.7-released.md]: Perfect! 🙂

**Review by alexandersandberg — APPROVED** [review]
Left one small comment, but other than that it's looking great now! 👏

File: _posts/2022-09-06-swift-5.7-released.md [hunk 13]
```diff
@@ -0,0 +1,161 @@
...
+  * The API digester and other tools have been added in the new smaller download.
+  * The SDK layout has been further streamlined by removing unnecessary path components.
+  * The developer libraries were restructured to allow multi-architectural installations as work continues to support ARM64 and X86 targets in addition to X64.
+  * Initial support for static Swift libraries was implemented for libraries other than the Swift standard library, and can be used with build tools such as CMake and Bazel on Windows.
+
+## Downloads
+
+Official binaries are [available for download](https://swift.org/download/) from [Swift.org](http://swift.org/) for Xcode, Windows, and Linux.
+
+You can also install RPMs for Amazon Linux 2 and CentOS 7 for **experimental use only**. Please provide your [feedback](https://bugs.swift.org/). Use the instructions below for RPM installation:
```
Comment (0xTim) [inline _posts/2022-09-06-swift-5.7-released.md:101]: This should probably point to GH now as JIRA isn't a thing?

**Review by 0xTim — APPROVED** [review]
Looks great! One comment about JIRA but not a blocker

File: _posts/2022-09-06-swift-5.7-released.md (hunk 13 above)
Comment (alexandersandberg) [reply _posts/2022-09-06-swift-5.7-released.md:101]: 👍 We do this here already: https://www.swift.org/download/#rpm

Comment (hborla) [reply _posts/2022-09-06-swift-5.7-released.md]: Fixed!

**Comment by hborla** [comment]
@swift-ci test

**Comment by hborla** [comment]
@swift-ci test

**Review by shahmishal — APPROVED** [review]
